src/presentations/Header.js

```diff
@@ -0,0 +1,136 @@
+        // this.doResizeActions = this.doResizeActions.bind(this);
+        // this.cartClick = this.cartClick.bind(this);
+        // this.resetCart = this.resetCart.bind(this);
+        // this.resize = this.resize.bind(this);
+    }
+    componentDidMount(){
+        if(this.props.signin.userData && this.props.signin.userData.userEmail){
+            this.setState({userName:this.props.signin.userData.userEmail});
+        } else if(this.props.signup && this.props.signup.userData && this.props.signup.userData.userEmail){
+            this.setState({userName:this.props.signup.userData.userEmail});
+        }
+        this.doResizeActions();
+        window.addEventListener('resize',this.resize,false);
+    }
+    componentWillUnmount(){
+        window.removeEventListener('resize',this.resize,false);
+    }
+
+    static getDerivedStateFromProps(props,state){
+        if(props.cartInfo){
+            state.itemCount= props.cartInfo.length;
+        }
+        if(props.productInfo && props.productInfo.cartItems){
+            state.itemCount= props.productInfo.cartItems.length
+        }
+        return state;
+    }
+    resize=()=>{
+        this.doResizeActions();
+    }
+    resetCart=()=>{
+        this.props.resetCartData();
+    }
+    doResizeActions=()=>{
+        const dimensions = getViewPortDimensions();
+        let ismobileview;
+        if(dimensions.width <= 480){
+            ismobileview = true;
+        } else{
+            ismobileview = false;
+        }
+        this.setState({dimensions: dimensions,isMobileView:ismobileview});
+    }
+    cartClick=()=>{
+        if(this.state.dimensions && this.state.dimensions.width && this.state.dimensions.width < 480){ //1024
+            this.props.router.push('/cart');
+        } else {
+            this.props.cartClick();
+        }
+    }
+    clickEvent=(event)=>{}
+    showHome=()=>{
+        this.props.router.push('/');
+    }
+    render(){   
+        const {userName,itemCount} = this.state;
+        return (<div className="header-area">
+            <header>
+                <nav className="sticky-nav">
+                    <div className="row">
+                        <img onClick={this.showHome} src={window.location.origin + '/static/images/logo.png'} alt="logo" className="logo" /> {/* {Logo} */}
+                        <ul className="main-nav js--main-nav">
+                            {!this.state.isMobileView && (<React.Fragment>
+                            <li><Link to={`/home`} >Home</Link></li>
+                            <li><Link to="/products">Products</Link></li>
+                            </React.Fragment>)}
+                            {this.state.isMobileView && (<React.Fragment>
+                            <li><Link to={`/home`} >H</Link></li>
+                            <li><Link to="/products">P</Link></li>
+                            <li><Link to={`/signin`} >U</Link></li>
+                            <li><Link to="/signup">NU</Link></li>
```

Comment:
can  you use media queries to design header links for mobile and desktop views?
What is H P U NU?